### ISSUE TITLE
Input selector node, output selector node

### ISSUE THREAD

**Issue by zopieux** [chaiNNer-org/chaiNNer#1465]
**Motivation**
It's painful to switch between different inputs (resp. outputs) of a given subgraph in chaiNNer, in other word provide alternative, "parallel" implementations of some processing and then forward *just one of them* down (resp. up) the processing graph.

**Description**
These nodes are basically parametric passthroughs with arbitrary inputs/ouputs, similar to how *Text pattern* can have a dynamic number of inputs.

* The input selector can receive an artbirary number of inputs (type: X, any) plus an index (type: numeric). It only consumes (enables) the specified input and passes it through as its output (type: X). **This is useful to quickly switch between different upstream processing subgraphs**, for example different input images or models, in case I don't need/want to iterate.
* The output selector receives one input (type: X, any) plus an index (type: numeric) and has an arbitrary number of outputs (type: X). It only produces (enables) the specific output index and passes its input through to it. **This is useful to quickly switch between different downstream processing subgraphs**, for example switch between "save image" and "preview image".

![graphviz(1)](https://user-images.githubusercontent.com/81353/211438501-e1625d36-1e4e-452b-94f0-02f9f979723a.svg)

**Alternatives**
Status quo: one has to painfully enable/disable and reroute N inputs to 1 subgraph (resp. 1 input to M subgraphs).

**Comment by joeyballentine** [comment]
Can't you use the passthrough node to route a single input to multiple places so you only have to swap the one input to the passthrough node?

**Comment by zopieux** [comment]
> Can't you use the passthrough node to route a single input to multiple places so you only have to swap the one input to the passthrough node?

Yes, this is basically what I've been doing to keep my sanity, to better organize my edges. I don't see how this solves the constant need for rerouting however, am I missing something? In other words, I'd very much prefer to type a number than having to reconnect an edge. Additionally, since the index would be an input, one could even reuse it in multiple selectors.

**Comment by joeyballentine** [comment]
> Additionally, since the index would be an input, one could even reuse it in multiple selectors.

This is a great point actually. You've convinced me :)
